### PR TITLE
Add timestamp and specs

### PR DESCRIPTION
**Pull request by Madpaia**
Adds timestamp to wme

### CODE REVIEW

File: lib/retex.ex [hunk 1]
```diff
@@ -11,15 +22,19 @@ defmodule Retex do
...
+  @spec new :: Retex.t()
   def new() do
     %{graph: graph} = %Retex{}
     graph = Graph.add_vertex(graph, Retex.Root.new())
     %Retex{graph: graph}
   end
 
+  @spec add_wme(Retex.t(), Retex.Wme.t()) :: Retex.t()
   def add_wme(%Retex{} = network, %Retex.Wme{} = wme) do
+    wme = Map.put(wme, :timestamp, :os.system_time(:seconds))
```
Comment (lorenzosinisi) [inline lib/retex.ex:37]: what about using put_new here so that if the value is already given it doesn't get overwritten? What do you think?

**Review by lorenzosinisi — COMMENTED** [review]
Thank you @Madpaia, I just have one question :)

File: lib/retex.ex (hunk 1 above)
Comment (Madpaia) [reply lib/retex.ex:37]: the timestamp is only set here during addition, so i would leave it, and eventually overwrite to ensure consistent order. What do you think?

**Comment by lorenzosinisi** [comment]
Good point ill leave it @Madpaia ! Thanks!

**Pull request merged**